Documentation/disaster-recovery.md

```diff
@@ -106,3 +106,107 @@ kubectl create -f operator.yaml
+
+1. Make sure the old Kubernetes cluster is completely torn down and the new Kubernetes cluster is up and running without rook.
+1. Backup `/var/lib/rook` in all the rook nodes to a different directory. Backups will be used later.
+1. Pick a `/var/lib/rook/rook-ceph/rook-ceph.config` from any previous Rook Ceph node and save the old cluster `fsid` from its content.
+1. Remove `/var/lib/rook` from all the Rook Ceph nodes.
+1. Add identical `CephCluster` descriptor to the new Kubernetes cluster, especially identical `spec.storage.config` and `spec.storage.nodes`, except `mon.count`, which sets to `1`.
+1. Add identical `CephFilesystem` `CephBlockPool` `CephNFS` `CephObjectStore` descriptors (if any) to the new Kubernetes cluster.
+1. Install Rook Ceph in the new Kubernetes cluster.
+1. Watch the operator logs with `kubectl logs -f rook-ceph-operator-xxxxxxx`, and wait until the orchestration has settled.
+1. **STATE:** Now the cluster will have `rook-ceph-mon-a`, `rook-ceph-mgr-a`, and all the auxiliary pods up and running, and zero (hopefully) `rook-ceph-osd-X-xxxxxx` running. Rook should not start any OSD daemon since all devices belongs to the old cluster (which have a different `fsid`).
+1. Run `kubectl exec -it rook-ceph-mon-a-XXXXXX bash` to enter the `rook-ceph-mon-a` pod,
+
+    ```shell
+    mon-a# cat /etc/ceph/keyring-store/keyring  # save this keyring content for later use
+    mon-a# exit
+    ```
+
+1. Stop the Rook operator by running `kubectl edit deploy/rook-ceph-operator` and set `replicas` to 0.
+1. Stop cluster daemons by running `kubectl delete deploy/X` where X is every deployment in namespace `rook-ceph`, except `rook-ceph-operator` and `rook-ceph-tools`.
+1. Save the `rook-ceph-mon-a` address with `kubectl get cm/rook-ceph-mon-endpoints -o yaml` in the new Kubernetes cluster for later use.
+
+1. SSH to the host where `rook-ceph-mon-a` in the new Kubernetes cluster resides.
+    1. Remove `/var/lib/rook/mon-a`
+    2. Pick a healthy `rook-ceph-mon-X` directory (`/var/lib/rook/mon-X`) in the previous backup, copy to `/var/lib/rook/mon-a`. `X` is any healthy mon node of the old cluster.
+    3. Replace `/var/lib/rook/mon-a/keyring` with the saved keyring, preserving only the `[mon.]` section, remove `[client.admin]` section.
+    4. Run `docker run -it --rm -v /var/lib/rook:/var/lib/rook ceph/ceph:v14.2.1-20190430 bash`. The Docker image tag should match the Ceph version used in the Rook cluster.
+
+        ```shell
+        container# cd /var/lib/rook
+        container# ceph-mon --extract-monmap m --mon-data ./mon-a/data  # Extract monmap from old ceph-mon db and save as m
+        container# monmaptool --print m  # Print the monmap content, which reflects the old cluster ceph-mon configuration.
+        container# monmaptool --rm a m  # Delete `a` from monmap.
+        container# monmaptool --rm b m  # Repeat this until all the old ceph-mons are removed
```

Comment:
For users less knowledgeable about Ceph, I think this could be improved somewhat.
`# Repeat, and delete `b` from the monmap`

Then on the next line 
`# Repeat this pattern until all the old ceph-mons are removed`

Reply:
fixed